fuel-vm/src/checked_transaction/types.rs

```diff
@@ -401,3 +402,79 @@ pub mod upload {
         }
     }
 }
+
+/// For [`fuel_tx::Blob`]
+pub mod blob {
+    use super::super::{
+        balances::{
+            initial_free_balances,
+            AvailableBalances,
+        },
+        Checked,
+        IntoChecked,
+    };
+    use crate::checked_transaction::{
+        CheckError,
+        NonRetryableFreeBalances,
+    };
+    use fuel_tx::{
+        AssetId,
+        Blob,
+        Cacheable,
+        Chargeable,
+        ConsensusParameters,
+        FormatValidityChecks,
+    };
+    use fuel_types::BlockHeight;
+
+    /// Metadata produced by checking [`fuel_tx::Blob`].
+    #[derive(Debug, Clone, Eq, PartialEq, Hash)]
+    pub struct CheckedMetadata {
```

Comment:
This struct seems to repeat with same contents for almost all tx types does it make sense to have separate copy for each?